doc/tui_rstgen.py

```diff
@@ -0,0 +1,68 @@
+"""Provides a module for generating Fluent TUI result files."""
+
+import fnmatch
+import os
+from pathlib import Path
+import shutil
+
+_THIS_DIRNAME = os.path.dirname(__file__)
+
+
+def _get_tui_docdir(mode: str):
+    return os.path.normpath(
+        os.path.join(
+            _THIS_DIRNAME,
+            "..",
+            "doc",
+            "source",
+            "api",
+            mode,
+            "tui",
+        )
+    )
+
+
+def _get_tui_filepath(mode: str):
+    return os.path.normpath(
+        os.path.join(
+            _THIS_DIRNAME,
+            "..",
+            "src",
+            "ansys",
+            "fluent",
+            "core",
+            f"{mode}",
+        )
+    )
+
+
+def _get_tui_file_name(mode: str):
+    for file in os.listdir(_get_tui_filepath(mode)):
+        if fnmatch.fnmatch(file, "tui_*.py"):
+            return file
+
+
+def generate():
```

Comment:
Does this work even if we don't generate the hierarchy of .rst files/directories?

Reply:
There is only single file inside each of `doc/source/api/meshing/tui/index.rst` and `doc/source/api/solver/tui/index.rst` therefore we are creating that single `index.rst` file using Python.  

We don't have hierarchy of .rst files/directories for tui, it is for datamodel only.